torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -375,21 +402,20 @@ void runPreAutodiffPassPipeline(std::shared_ptr<Graph>& graph) {
       "After CheckInplace (end of runPreAutodiffPassPipeline)\n", *graph);
 }
 
-FusionBehavior getCurrentBehavior(size_t remaining_depth) {
+FusionBehavior ProfilingGraphExecutorImpl::getCurrentBehavior(size_t remaining_depth) {
```

Comment:
Is it possible to also update getBailoutDepth (line ~130) to be a method in ProfilingGraphExecutorImpl and for it to use the local strategy? Didn't look too closely at it but it seems like it's possible

The issue I was imagining was:
1) ProfilingGraphExecutorImpl is initialized
2) fusion strategy changes
3) [getPlanFor](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/graph_executor.cpp#L411) gets called with the *new* number of bailouts, which might be different (which would cause incorrect indexing into the fusion strategies)

Reply:
This got kind of hairy, so I'm doing it as a stacked PR.

Reply:
Let me know if it would be easier if they were merged